### PR TITLE
Remove problem dependency

### PR DESCRIPTION
**Pull request by jbaiera**
Snapshot builds and CI is failing because a dependency cannot be reliably pulled from the repository it is in. The locations it is persisted in are either inaccessible or being sunset, and we luckily do not even use it, so this PR removes it from being considered by the build.

### CODE REVIEW

**Comment by jbaiera** [comment]
Opened against master in error

File: buildSrc/src/main/groovy/org/elasticsearch/hadoop/gradle/BuildPlugin.groovy [hunk 1]
```diff
@@ -172,7 +172,7 @@ class BuildPlugin implements Plugin<Project>  {
         project.repositories.mavenCentral()
         project.repositories.maven { url "https://conjars.org/repo" }
         project.repositories.maven { url "https://clojars.org/repo" }
-        project.repositories.maven { url 'https://repo.spring.io/plugins-release' }
+        project.repositories.maven { url 'https://repo.spring.io/plugins-release-local' }
```
Comment (mark-vieira) [inline buildSrc/src/main/groovy/org/elasticsearch/hadoop/gradle/BuildPlugin.groovy:175]: Why was this required _in addition_ to dropping the dependency?

**Review by mark-vieira — APPROVED** [review]

File: buildSrc/src/main/groovy/org/elasticsearch/hadoop/gradle/BuildPlugin.groovy (hunk 1 above)
Comment (jbaiera) [reply buildSrc/src/main/groovy/org/elasticsearch/hadoop/gradle/BuildPlugin.groovy]: I've reverted this. It doesn't seem like we actually make use of this repository at all, especially since as it is it requires authorization to access. I'll put out a follow up to clean it out.
